### PR TITLE
[CI] Fix reset connection when downloading dependencies

### PR DESCRIPTION
**Pull request by v1v**
https://github.com/elastic/apm-agent-java/pull/1377 did not solve the issue with the reset connection.

Let's try more maven magic tricks to fix the issue

Consequences: build time gets increased by 2.5 times:
- Used to be->  6 mnts average in the CI
- With this -> 14 mnts average in the CI

## References

- [here](https://stackoverflow.com/questions/55899091/maven-retry-dependency-download-if-failed)

## Issues

Relates to https://github.com/elastic/apm-agent-java/issues/1375

### CODE REVIEW

**Review by kuisathaverat — APPROVED** [review]

**Review by SylvainJuge — COMMENTED** [review]
I think that we should be able to only keep `-Dhttp.keepAlive=false` as it's the only parameter that provided any difference as tested with #1393 thus all those extra values should be removed `-Dhttps.protocols=TLSv1.2 -Dmaven.wagon.http.retryHandler.count=3 -Dmaven.wagon.http.pool=false`

**Review by cachedout — APPROVED** [review]

**Comment by v1v** [comment]
jenkins run the tests please

**Comment by v1v** [comment]
jenkins run the tests please

File: Jenkinsfile [hunk 1]
```diff
@@ -31,7 +31,7 @@ pipeline {
     issueCommentTrigger('(?i).*(?:jenkins\\W+)?run\\W+(?:the\\W+)?(?:benchmark\\W+)?tests(?:\\W+please)?.*')
   }
   parameters {
-    string(name: 'MAVEN_CONFIG', defaultValue: '-B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -Dhttps.protocols=TLSv1.2', description: 'Additional maven options.')
+    string(name: 'MAVEN_CONFIG', defaultValue: '-V -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -Dhttps.protocols=TLSv1.2 -Dmaven.wagon.http.retryHandler.count=3 -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false', description: 'Additional maven options.')
```
Comment (v1v) [inline Jenkinsfile:34]:
```suggestion
    string(name: 'MAVEN_CONFIG', defaultValue: '-V -B -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -Dhttp.keepAlive=false', description: 'Additional maven options.')
```

**Comment by v1v** [comment]
@SylvainJuge , I just committed https://github.com/elastic/apm-agent-java/pull/1399/commits/66b45f0d2c86c198662c9b1e19c1c5a486e464dc  and I'll run a few builds in this PR to validate we don't suffer any reset connections

**Review by SylvainJuge — APPROVED** [review]
Thanks, we can merge it as-is for now, as investigating network settings might take a bit more time.

**Comment by SylvainJuge** [comment]
Well, there still seems to be unexpected downloads with `http` instead of `https` :sob:

**Comment by v1v** [comment]
I'll revert https://github.com/elastic/apm-agent-java/commit/66b45f0d2c86c198662c9b1e19c1c5a486e464dc

**Pull request merged**